swim/ping_request_sender.go

```diff
@@ -82,6 +82,9 @@ func (p *pingRequestSender) SendPingRequest() (*pingResponse, error) {
 	var res pingResponse
 	select {
 	case err := <-p.MakeCall(ctx, &res):
+		if err == nil {
+			p.node.memberlist.Update(res.Changes)
```

Comment:
This was handled in the pingNextMember before, which worked since thats the only caller of sendPingRequests. Handling it there is an accident waiting to happen if we would ever send ping-reqs for another reason. I believe it's should be part of this diff because it's part of refactoring the ping-reqs mechanism.